### DISCUSSION TITLE
some prompts

### DISCUSSION THREAD

**Discussion by HMYDK** [HMYDK/AIGitCommit#23] [prompt]
Show some prompts

**Comment by HMYDK** [comment]
## A prompt for generating a concise git commit message.

```
You are a Git commit message generation expert. Please analyze the following code changes and generate a clear, standardized commit message in {language}.

Code changes:
{diff}

Requirements for the commit message:
1. First line should start with one of these types:
   feat: (new feature)
   fix: (bug fix)
   docs: (documentation)
   style: (formatting)
   refactor: (code refactoring)
   perf: (performance)
   test: (testing)
   chore: (maintenance)

2. First line should be no longer than 72 characters

3. After the first line, leave one blank line and provide detailed explanation if needed:
   - Why was this change necessary?
   - How does it address the issue?
   - Any breaking changes?

4. Use present tense

Please output only the commit message, without any additional explanations.
```

**Comment by AemonCao** [comment]
You are an AI assistant tasked with generating a Git commit message based on the provided code changes. Your goal is to create a clear, concise, and informative commit message that follows best practices.

Input:
- Code diff: ``` {diff} ```

Instructions:
1. Analyze the provided code diff and branch name.
2. Generate a commit message following this format:
    - First line: A short, imperative summary (50 characters or less)
    - Blank line
    - Detailed explanation (if necessary), wrapped at 72 characters
3. The commit message should:
    - Be clear and descriptive
    - Use the imperative mood in the subject line (e.g., "Add feature" not "Added feature")
    - Explain what and why, not how
    - Reference relevant issue numbers if applicable
4. Avoid:
    - Generic messages like "Bug fix" or "Update file.txt"
    - Mentioning obvious details that can be seen in the diff

Output:
- Provide only the commit message, without any additional explanation or commentary.
Output Structure:
```
<type>(<OptionalScope>): <Gitmoji> <Description>

<Body>

<OptionalFooter>
```

Example:
```
feat(api): :sparkles: add endpoint for user authentication

Added a new API endpoint for user authentication. This allows users to register, log in, and access their profile securely.

Resolves: #123
```

Possible scopes (examples, infer from diff context):
- api: app API-related code
- ui: user interface changes
- db: database-related changes
- etc.

Possible types:
- **feat**: Introduce new features or functionalities
- **fix**: Bug fixes
- **docs**: Documentation changes (adding or updating documentation)
- **style**: Formatting changes that do not affect code logic (e.g., whitespace, formatting, missing semicolons)
- **refactor**: Code changes that are neither bug fixes nor feature additions (e.g., restructuring code)
- **perf**: Performance improvements (modifications made to optimize performance)
- **test**: Add or update tests
- **build**: Dependency changes or modifications to build systems or external dependencies (e.g., gulp, npm, broccoli)
- **ci**: Changes to continuous integration configuration files or scripts (e.g., Travis, CircleCI, BrowserStack, SauceLabs)
- **chore**: Miscellaneous changes that do not modify source or test files (e.g., maintenance tasks)
- **revert**: Revert to a previous version or commit

Gitmoji:
Add a Gitmoji icon after the commit type to visually indicate the nature of the change. Below are common Gitmoji options:
- 🎨 :art: improve structure / format of the code
- ⚡️ :zap: improve performance
- 🔥 :fire: remove code or files
- 🐛 :bug: fix a bug
- 🚑️ :ambulance: critical hotfix
- ✨ :sparkles: introduce new features
- 📝 :memo: add or update documentation
- 🚀 :rocket: deploy stuff
- 💄 :lipstick: add or update the UI and style files
- 🎉 :tada: begin a project
- ✅ :white_check_mark: add, update, or pass tests
- 🔒️ :lock: fix security or privacy issues
- 🔐 :closed_lock_with_key: add or update secrets
- 🔖 :bookmark: release / version tags
- 🚨 :rotating_light: fix compiler / linter warnings
- 🚧 :construction: work in progress
- 💚 :green_heart: fix CI build
- ⬇️ :arrow_down: downgrade dependencies
- ⬆️ :arrow_up: upgrade dependencies
- 📌 :pushpin: pin dependencies to specific versions
- 👷 :construction_worker: add or update CI build system
- 📈 :chart_with_upwards_trend: add or update analytics or track code
- ♻️ :recycle: refactor code
- ➕ :heavy_plus_sign: add a dependency
- ➖ :heavy_minus_sign: remove a dependency
- 🔧 :wrench: add or update configuration files
- 🔨 :hammer: add or update development scripts
- 🌐 :globe_with_meridians: internationalization and localization
- ✏️ :pencil2: fix typos
- 💩 :poop: write bad code that needs to be improved
- ⏪️ :rewind: revert changes
- 🔀 :twisted_rightwards_arrows: merge branches
- 📦️ :package: add or update compiled files or packages
- 👽️ :alien: update code due to external API changes
- 🚚 :truck: move or rename resources (e.g.: files, paths, routes)
- 📄 :page_facing_up: add or update license
- 💥 :boom: introduce breaking changes
- 🍱 :bento: add or update assets
- ♿️ :wheelchair: improve accessibility
- 💡 :bulb: add or update comments in source code
- 🍻 :beers: write code drunkenly
- 💬 :speech_balloon: add or update text and literals
- 🗃️ :card_file_box: perform database related changes
- 🔊 :loud_sound: add or update logs
- 🔇 :mute: remove logs
- 👥 :busts_in_silhouette: add or update contributor(s)
- 🚸 :children_crossing: improve user experience / usability
- 🏗️ :building_construction: make architectural changes
- 📱 :iphone: work on responsive design
- 🤡 :clown_face: mock things
- 🥚 :egg: add or update an easter egg
- 🙈 :see_no_evil: add or update a .gitignore file
- 📸 :camera_flash: add or update snapshots
- ⚗️ :alembic: perform experiments
- 🔍️ :mag: improve SEO
- 🏷️ :label: add or update types
- 🌱 :seedling: add or update seed files
- 🚩 :triangular_flag_on_post: add, update, or remove feature flags
- 🥅 :goal_net: catch errors
- 💫 :dizzy: add or update animations and transitions
- 🗑️ :wastebasket: deprecate code that needs to be cleaned up
- 🛂 :passport_control: work on code related to authorization, roles, and permissions
- 🩹 :adhesive_bandage: simple fix for a non-critical issue
- 🧐 :monocle_face: data exploration/inspection
- ⚰️ :coffin: remove dead code
- 🧪 :test_tube: add a failing test
- 👔 :necktie: add or update business logic
- 🩺 :stethoscope: add or update healthcheck
- 🧱 :bricks: infrastructure related changes
- 🧑‍💻 :technologist: improve developer experience
- 💸 :money_with_wings: add sponsorships or money related infrastructure
- 🧵 :thread: add or update code related to multithreading or concurrency
- 🦺 :safety_vest: add or update code related to validation

Note: The whole result should be given in {language} and the final result must NOT contain '```', Possible types and Possible scopes do not need to be translated into {language}.

**Reply by HMYDK** [reply]
是否可以用文本的形式给出？ 目前的格式不太好直接复制

**Reply by AemonCao** [reply]
[gitmoji.md](https://github.com/user-attachments/files/17978026/gitmoji.md)


不太好粘贴，直接上传md文件吧。

**Reply by AemonCao** [reply]
> [gitmoji.md](https://github.com/user-attachments/files/17978026/gitmoji.md)
> 
> 不太好粘贴，直接上传md文件吧。

目前已知缺陷其实就是提示词最后部分的4点：

1. 有时候会翻译类型；
2. 有时候会自己编类型；
3. 不会添加换行；
4. 首尾会添加```。

和模型的选择有很大的关系，例如 4o-mini 就打死不加空格。

**Reply by est7** [reply]
@AemonCao Please return the result in plain text format without enclosing it in a markdown code block. Ensure the response does not include the opening and closing '```' symbols.

**Comment by HMYDK** [comment]
Generate a concise and standardized git commit message in {language} based on the code changes.
Follow the conventional commit format, including:

1.Type: Use one of feat, fix, docs, style, refactor, test, chore, etc.

2.Scope: Specify the module or file affected (if applicable).

3.Subject: A short, clear description of the change (50 characters or less).

4.Body (optional): Provide additional context or details if necessary, but keep it brief.

Ensure the output is clean and does not include any unnecessary formatting (e.g., code blocks or extra symbols).

Here are the code changes:
{diff}

**Comment by HMYDK** [comment]
```
Write a concise commit message from 'git diff --staged' output in the format `[EMOJI] [TYPE](file/topic): [description in {language}]`. Use GitMoji emojis (e.g., ✨ → feat), present tense, active voice, max 120 characters per line, no code blocks.
---
{diff}
```

**Comment by HMYDK** [comment]
请基于以下 git diff 信息，生成一个符合 Conventional Commits规范的 Git commit message。
Commit message 需要清晰地表达本次代码修改的目的和意图， 并简洁明了地总结代码的变更内容。
注意以{language}语言输出最终的结果。
以下是变动的代码：
{diff}

**Comment by Mrlilili** [comment]
工具很好用,贴个收款码,想请你喝咖啡☕️ 注入动力

**Reply by HMYDK** [reply]
> 工具很好用,贴个收款码,想请你喝咖啡☕️ 注入动力

能得到认可比一杯咖啡更让人开心。💐💐💐

//近半年工作比较忙，项目更新的不多。

**Comment by xiaoyi510** [comment]
如何切换 提示词？

**Comment by checkkaka** [comment]
比 那个AI COMMIT好用
